contracts/StakeManager.sol

```diff
@@ -157,12 +180,32 @@ contract StakeManager is Ownable {
                 revert StakeManager__InvalidLockTime();
             }
         }
+
         _mintBonusMP(account, deltaTime, _amount);
 
+        //mp estimation
+        uint256 mpPerEpoch = _getMPToMint(_amount, EPOCH_SIZE);
+        if(mpPerEpoch < 1){
+            revert StakeManager__StakeIsTooLow();
+        }
+        uint256 thisEpochExpiredMP = _getMPToMint(_amount, block.timestamp - epochs[currentEpoch].startTime);
+        currentEpochExpiredMP += thisEpochExpiredMP; //TODO: SHIFT ESTIMATION TO LAST EPOCH
+        totalMPPerEpoch += mpPerEpoch;
+        uint256 maxMpToMint = _getMPToMint(_amount, MAX_BOOST * YEAR);
+        uint256 mpMaxBoostLimitEpochCount = maxMpToMint / mpPerEpoch;
+        uint256 mpMaxBoostLimitEpoch = currentEpoch + mpMaxBoostLimitEpochCount;
+        uint256 lastEpochAmountToMint = ((mpPerEpoch * (mpMaxBoostLimitEpochCount+1)) - maxMpToMint);
+        
+        expiredMPPerEpoch[mpMaxBoostLimitEpoch] += lastEpochAmountToMint; // some staked amount from the past
```

Comment:
This is confusing... If we just calculated `lastEpochAmountToMint` ("the amount to mint in the last epoch that this account would get MP"), why are we adding this to `expired`MPPerEpoch?

Reply:
Because when account reaches MAX_BOOST is not exactly at one epoch... Usually in that epoch, it would mint only 20% or whatever of its normal per epoch growth, and than mint the remainder in the subsequent epoch, to then start minting zero for the end of days.